src/ProductConstructionService/ProductConstructionService.Api/Configuration/PcsConfiguration.cs

```diff
@@ -49,6 +48,8 @@ public static string GetRequiredValue(this IConfiguration configuration, string
             builder.Configuration.AddAzureKeyVault(keyVaultUri, credential);
         }
 
+        string databaseConnectionString = builder.Configuration.GetRequiredValue(PcsConfiguration.DatabaseConnectionString);
```

Comment:
I don't follow what is the difference. Either we call the `builder.Configuration.GetRequiredValue` in `Program.cs` or just a few lines below in `ConfigurePcs`.

How come it starts working suddenly?

Reply:
The reason why it's failing is because we're trying to read a secret before adding the `KeyVault` in `ConfigurePcs`. This PR makes it so we add the KeyVault, then try to read the secret

Reply:
Ah, I see. GitHub folded those few lines for me